app/services/todopoint_mover_form.rb

```diff
@@ -0,0 +1,46 @@
+class TodopointMoverForm
+  include ActiveModel::Model
+
+  attr_accessor :list_id, :new_list_id, :todopoint_id
+
+  validate :wrong_board_list
+
+  def save
+    return false unless valid?
+    below_moved_todopoint
+    todopoint.update!(list_id: new_list_id, position: List.find(new_list_id).todopoints.size + 1)
+    true
```

Comment:
when you do multiple inserts or updates to the database consider using transactions. Especially when one failed operation damages data consistency, like in your case. Please read https://medium.com/@kristenrogers.kr75/rails-transactions-the-complete-guide-7b5c00c604fc